dtspec/api.py

```diff
@@ -244,7 +244,12 @@ def _parse_spec(self, json_spec):
 
     def generate_sources(self):
         "Used to generate all source data that will be passed back to the user"
-        for _scenario_name, scenario in self.spec["scenarios"].items():
+        for _scenario_name in self.spec["scenarios"].keys():
+            self.generate_sources(_scenario_name)
+
+    def generate_sources(self, scenario_name):
```

Comment:
This is the same name as the method above, so now I'm guessing `generate_sources` won't work as it did before.  Maybe call this `generate_sources_for_scenario`.